### ISSUE TITLE
`--project=@script` is broken or NEWS entry is incorrect (Julia 1.11.0-rc3)

### ISSUE THREAD

**Issue by StevenWhitaker** [JuliaLang/julia#55606]
This issue has two parts.

# Issue 1

The following NEWS entry is incorrect:
> If no path is given after (i.e. `--project=@script`) then (like `--project=@.`) the directory and its parents are searched for a Project.toml

In particular, it appears `--project=@script` has the same behavior as `--project=@.`, which doesn't seem intended (though maybe I'm reading the NEWS entry incorrectly).

## MWE

In `<current_directory>/script_test/test.jl`:
```julia
println("Current directory: ", pwd())
println("Active project: ", Base.active_project())
```

Result:
```bash
$ julia +1.11 --project=@script script_test/test.jl
Current directory: <current_directory>
Active project: <current_directory>/Project.toml # Expected: <current_directory>/script_test/Project.toml
```

See #55069 for more details.

# Issue 2

The previous part of the same NEWS entry is also incorrect:
> The `--project` argument now accepts `@script` to give a path to a directory with a Project.toml relative to the passed script file. `--project=@script/foo` for the `foo` subdirectory.

In particular, `--project=@script/script_test` activates `/script_test/Project.toml`. (Note this is an absolute path, not a relative path!)

## MWE

Same `test.jl` file as above.

Result:
```bash
$ julia +1.11 --project=@script/foo script_test/test.jl
Current directory: <current_directory>
Active project: /script_test/Project.toml # Expected: <current_directory>/script_test/foo/Project.toml
```

**Comment by tecosaur** [comment]
Thanks for reporting! Let's focus on #55069 though 🙂

**Comment by StevenWhitaker** [comment]
@tecosaur I agree that Issue 1 above is a dupe of #55069. However, #55069 doesn't mention anything about the `--project=@script/sub_dir` functionality (what I call Issue 2 above). So this issue isn't a complete duplicate. Would it be better to open a new issue with just Issue 2 above?

**Comment by tecosaur** [comment]
At a glance, I suspect that Issue 2 might be solved by fixing Issue 1 (if the base path is determined incorrectly, paths relative to it will be incorrect too).

**Comment by StevenWhitaker** [comment]
I am slightly doubtful the same fix will apply to both issues because in Issue 2 above an absolute path somehow gets introduced. In other words, if Issue 2 did just depend on Issue 1 getting the correct base path, then the Issue 2 result should be
```bash
Active project: <current_directory>/foo/Project.toml
```
which isn't the case, as I report above (`foo` doesn't even show up in the active project).

But I guess we'll see.

**Comment by tecosaur** [comment]
Ah right, I think I just missed that nuance in a skim-read. I'll reopen the issue 🙂
